packages/airnode-node/src/providers/state.ts

```diff
@@ -67,14 +79,21 @@ export function scrub<T>(state: ProviderState<T>): ProviderState<T> {
   return removeKeys(state, ['config', 'masterHDNode', 'provider']) as ProviderState<T>;
 }
 
-export function refresh(state: ProviderState<any>) {
+export function refresh<T extends EVMProviderState>(state: ProviderState<T>) {
   if (state.settings.chainType === 'evm') {
     // The serverless function does not return an instance of an Ethereum
     // provider, so we create a new one before returning the state
-    const masterHDNode = evm.getMasterHDNode(state.config);
+    const masterHDNode = evm.getMasterHDNode(state.config!);
     const provider = evm.buildEVMProvider(state.settings.url, state.settings.chainId);
-    return update(state, { masterHDNode, provider }) as ProviderState<EVMProviderState>;
+    return update(state, { masterHDNode, provider } as Partial<ProviderState<T>>);
   }
 
   throw new Error(`Unknown chain type: ${state.settings.chainType}`);
 }
+
+export function splitStatesBySponsorAddress(providerStates: ProviderStates): ProviderState<EVMProviderSponsorState>[] {
+  return flatMap(providerStates.evm, (providerState) => {
+    const groupedRequests = groupRequestsBySponsorAddress(providerState.requests);
```

Comment:
Nit: I'd slighly prefer if the function was implemented like this:
```
export function groupRequestsBySponsorAddress(requests: GroupedRequests) {
  const apiCalls = requests.apiCalls.filter((a) => !!a.sponsorAddress);
  const withdrawals = requests.withdrawals.filter((w) => !!w.sponsorAddress);

  const apiCallsBySponsorAddress = groupBy(apiCalls, 'sponsorAddress');
  const withdrawalsBySponsorAddress = groupBy(withdrawals, 'sponsorAddress');

  const uniqueSponsorAddresses = mapUniqueSponsorAddresses(requests);

  const groupedRequests = uniqueSponsorAddresses.reduce((acc, sponsorAddress) => {
    const requests = {
      apiCalls: apiCallsBySponsorAddress[sponsorAddress!] || [],
      withdrawals: withdrawalsBySponsorAddress[sponsorAddress!] || [],
    };

    return { ...acc, [sponsorAddress]: requests };
  }, {} as RequestsBySponsorAddress);

  return groupedRequests;
}
```

since TS treats `{}` basically as any. This is similar why this:
```
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface Withdrawal {}
```
doesn't really work.